vscode-extensions/vscode-spring-boot/lib/explorer/structure-tree-manager.ts

```diff
@@ -17,14 +16,20 @@ export class StructureManager {
             if (node && node.getReferenceValue) {
                 const reference = node.getReferenceValue();
                 if (reference) {
-                    // Reference is a specific URL that should be passed to java.open.file command
-                    commands.executeCommand('java.open.file', reference);
+                    const uri = Uri.parse(reference);
+                    if (uri.scheme === 'jdt') {
+                        // Reference is a specific URL that should be passed to java.open.file command
+                        commands.executeCommand('java.open.file', reference);
+                    } else {
+                        // Reference is a specific URL that should be passed to vscode.open command
+                        commands.executeCommand('vscode.open', uri);
+                    }
                 }
             }
         }));
 
         context.subscriptions.push(commands.registerCommand("vscode-spring-boot.structure.grouping", async (node: StereotypedNode) => {
-            const projectName = node.getProjectId();
+            const projectName = node.getNodeId();
```

Comment:
I introduced this change to use the project ID instead of the node ID here:
https://github.com/spring-projects/spring-tools/commit/2702658cfcdc9ba2a7d2c6d882b5b4a3c222ff12

I think we need to use the project ID, since the node ID could be something that is a very special label. For example for restbucks (when using Modulith), the label for the project looks something like `restbucks (de.odrotbohm.restbucks)` - which is not the exact project element name. Using the node ID (which includes the project label) fails here, since the project in the language server is called `restbucks`. Therefore I introduced (for project nodes) a dedicated attribute in the JSON to have the exact project name at hand and use that when identifying projects between the UI and the language server.